app/src/main/java/com/zulip/android/networking/AsyncEdit.java

```diff
@@ -0,0 +1,25 @@
+package com.zulip.android.networking;
+
+import com.zulip.android.ZulipApp;
+import com.zulip.android.activities.ZulipActivity;
+import com.zulip.android.models.Message;
+
+public class AsyncEdit extends ZulipAsyncPushTask {
+
+    /**
+     * Initialise an AsyncEdit task to edit a specific message.
+     *
+     * @param zulipActivity The calling Activity
+     * @param msg           The message to edit.
+     */
+    public AsyncEdit(ZulipActivity zulipActivity, Message msg, String messageContent) {
+        super((ZulipApp) zulipActivity.getApplication());
+
+        this.setProperty("content", messageContent);
+        this.setProperty("message_id", String.valueOf(msg.getID()));
+    }
+
+    public final void execute() {
+        execute("PATCH", "v1/messages");
```

Comment:
how about we use [retrofit 2](https://square.github.io/retrofit/) for it? See [ZulipServices](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/service/ZulipServices.java).